pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go

```diff
@@ -15,6 +15,7 @@
 package kubeapiserver
 
 import (
+	worker "github.com/gardener/gardener/extensions/pkg/controller/worker/genericactuator"
```

Comment:
I couldn't find any constant in g/g for this `v1beta1constants`.  So I used the one from the extension as I didn't want to introduce a new one if it goes untracked.
However as suggested by to add it there -- so this is fixed with the [commit](https://github.com/gardener/gardener/pull/5497/commits/2d60faf78e7644cddd6a3b424316b629a7ed483b)

Reply:
With GEP-01 (extensibility) cloudprovider specific details are extracted to extensions. gardenlet does not need to know anything about MCM or cannot make any assumption that MCM is used. Actually from gardenlet's point of view there is only the Worker resource and that's is the contract. The fact that provider extensions choose to deploy MCM as part of the Worker reconciliation is not a thing that gardenlet has to assume. IMO this PR is violating GEP-01 as it is making gardenlet to configure dependency-watchdog assuming that MCM is used. In theory, a provider extension can implement the contract without using MCM.

Reply:
@ialidzhikov See https://github.com/gardener/gardener/pull/5497#issuecomment-1055392531

Reply:
I agree to your concern @ialidzhikov. 
Do you suggest an alternate approach here or you are fine with adding MCM deployment name in the constants.

Reply:
We can think of it. Currently I didn't want to introduce new semantics of identifying a scale resources. As the implementation already requires us to provide MCM as part of  scaleRef as mentioned below
```
scaleRef:
       apiVersion: apps/v1
       kind: Deployment
       name: machine-controller-manager
``` 
So `scaleRefDependsOn` is not the issue here, if we want to do what you suggest we also need to change the design for `scaleRef` itself to have a new approach for selecting the deployment. 

> Do we need actually scaleRefDependsOn? Can't we simply scale down all components when the probe fails?

The logic currently works seamless for both ScaleUp and ScaleDowns.  
Like what you mentioned scalingDown is done all at once. But, we need to consider the semantics of ScaleUp here. This is where we have a problem today, we scale up everything at once which is detrimental to even introduce MCM additionally along with KCM. As once MCM comes ups it will mark the nodes as `Unknown` before giving KCM any chance to update the node status and will start removing them. To avoid this we wish to delay it's `scaleUpDelaySeconds`.  
However, even then we rung the risk of KCM not being available and just starting MCM with some delay without checking if KCM is up will lead us to the same issue. So we introduced  the `scaleRefDependOn` semantics to avoid running MCM on a state of the system which is not yet updated by KCM. 
To not reinvent the wheel `scaleRefDependsOn` is just an array of same type as `scaleRef`. 

We will explore if there is a better way to do it without breaking the extension contract for Gardener.

Reply:
I again suggest to be pragmatic here and rather focus on getting this change in so that we can move on the other (blocking) issues like https://github.com/gardener/dependency-watchdog/issues/36. IMO it's not problematic to specify MCM here as explained above.

Reply:
> We will explore if there is a better way to do it without breaking the extension contract for Gardener.

However, can you create an issue on dependency-watchdog side to make sure that we don't forget about this and this item is considered/worked on/brainstormed on when there is capacity? Thanks in advance!

Reply:
As suggested created an [Issue on DWD](https://github.com/gardener/dependency-watchdog/issues/42) to track this.